net-nds/389-ds-base/389-ds-base-1.4.4.4.ebuild

```diff
@@ -0,0 +1,261 @@
+			sys-libs/libselinux[python,${PYTHON_MULTI_USEDEP}]
+		')
+	)
+	systemd? ( >=sys-apps/systemd-244 )
+	"
+
+# perl is for mkDBErrStrs.pl being run in make step
+BDEPEND="dev-lang/perl
+	virtual/pkgconfig
+	${PYTHON_DEPS}
+	$(python_gen_cond_dep '
+		dev-python/argparse-manpage[${PYTHON_MULTI_USEDEP}]
+	')
+	doc? ( app-doc/doxygen )
+	test? ( dev-util/cmocka )
+"
+
+# perl dependencies are for logconv.pl
+RDEPEND="${DEPEND}
+	!dev-libs/svrcore
+	!net-nds/389-ds-base:0
+	acct-user/dirsrv
+	acct-group/dirsrv
+	${PYTHON_DEPS}
+	$(python_gen_cond_dep '
+		dev-python/pyasn1[${PYTHON_MULTI_USEDEP}]
+		dev-python/pyasn1-modules[${PYTHON_MULTI_USEDEP}]
+		dev-python/argcomplete[${PYTHON_MULTI_USEDEP}]
+		dev-python/python-dateutil[${PYTHON_MULTI_USEDEP}]
+		dev-python/python-ldap[sasl,${PYTHON_MULTI_USEDEP}]
+		dev-python/distro[${PYTHON_MULTI_USEDEP}]
+	')
+	virtual/perl-Archive-Tar
+	virtual/perl-DB_File
+	virtual/perl-IO
+	virtual/perl-Getopt-Long
+	virtual/perl-IO-Compress
+	virtual/perl-MIME-Base64
+	virtual/perl-Scalar-List-Utils
+	virtual/perl-Time-Local
+	virtual/logger
+	selinux? ( sec-policy/selinux-dirsrv )
+"
+
+distutils_enable_tests pytest
+
+src_prepare() {
+	# this is for upstream GitHub issue 4292
+	if use !systemd; then
+		sed -i \
+			-e 's|WITH_SYSTEMD = 1|WITH_SYSTEMD = 0|' \
+			Makefile.am || die
+	fi
+	eapply "${FILESDIR}/${PN}-db-gentoo.patch"
+	eapply_user
+
+	eautoreconf
+}
+
+src_configure() {
+	econf \
+		$(use_enable accountpolicy acctpolicy) \
+		$(use_enable bitwise) \
+		$(use_enable dna) \
+		$(use_enable pam-passthru) \
+		$(use_enable autobind) \
+		$(use_enable auto-dn-suffix) \
+		$(use_enable debug) \
+		$(use_enable ldapi) \
+		$(use_with selinux) \
+		$(use_with systemd) \
+		$(use_with systemd systemdsystemunitdir "$(systemd_get_systemunitdir)") \
+		$(use_with systemd systemdgroupname "dirsrv.target") \
+		$(use_with systemd tmpfiles-d "/usr/lib/tmpfiles.d") \
+		$(use_with !systemd initddir "/etc/init.d") \
+		$(use_enable test cmocka) \
+		--enable-rust \
+		--enable-rust-offline \
+		--with-pythonexec="${PYTHON}" \
+		--with-fhs \
+		--with-openldap \
+		--with-db-inc="$(db_includedir)" \
+		--disable-cockpit
+
+		rm "${S}"/.cargo/config || die
+}
+
+src_compile() {
+	export CARGO_HOME="${ECARGO_HOME}"
+	default
+	if use doc; then
+		doxygen "${S}"/docs/slapi.doxy || die
+	fi
+
+	cd "${S}"/src/lib389 || die
+	distutils-r1_src_compile
+
+	# argparse-manpage dynamic man pages have hardcoded man v1 in header
+	sed -i \
+		"1s/\"1\"/\"8\"/" \
+		"${S}"/src/lib389/man/ds{conf,ctl,idm,create}.8 || die
+}
+
+src_test () {
+	cd "${S}"/src/lib389 || die
+	distutils-r1_src_test
+}
+
+src_install() {
+	# -j1 is a temporary workaround for bug #605432
+	emake -j1 DESTDIR="${D}" install
+
+	# Install gentoo style init script
+	# Get these merged upstream
+	newinitd "${FILESDIR}"/389-ds.initd-r1 389-ds
+	newinitd "${FILESDIR}"/389-ds-snmp.initd 389-ds-snmp
+
+	dotmpfiles "${FILESDIR}"/389-ds-base.conf
```

Comment:
The DS installer creates files under tmpfiles dir with the name "dirsrv-$instance_name.conf" for each instance. For consistency it might be good to name this "dirsrv.conf", but it's not super important.